contracts/whitelist/src/contract.rs

```diff
@@ -284,6 +309,45 @@ pub fn execute_update_per_address_limit(
         .add_attribute("per_address_limit", per_address_limit.to_string()))
 }
 
+/// Increase member limit. Must include a fee if crossing 1000, 2000, etc member limit.
+pub fn execute_increase_member_limit(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    member_limit: u32,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+    if config.member_limit > member_limit || member_limit > MAX_MEMBERS {
+        return Err(ContractError::InvalidMemberLimit {
+            min: config.member_limit,
+            max: MAX_MEMBERS,
+            got: member_limit,
+        });
+    }
+
+    // if new limit crosses 1,000 members, requires upgrade fee. Otherwise, free upgrade.
+    let old_limit = Decimal::new(config.member_limit.into(), 3).ceil();
+    let new_limit = Decimal::new(member_limit.into(), 3).ceil();
+    let upgrade_fee: u128 = if new_limit > old_limit {
+        (new_limit - old_limit).to_u128().unwrap() * PRICE_PER_1000_MEMBERS
+    } else {
+        0
+    };
+    let payment = may_pay(&info, NATIVE_DENOM)?;
```

Comment:
This doesn't appear to be actually charging the fee by calling `burn_and_distribute()`.

Reply:
good catch!